framework/meta-lib/src/tools/wasm_extractor.rs

```diff
@@ -44,50 +54,136 @@ fn populate_wasm_info(
     wasm_data: Vec<u8>,
     extract_imports_enabled: bool,
     check_ei: &Option<EIVersion>,
+    mut view_endpoints: HashMap<&str, usize>,
 ) -> Result<WasmInfo, BinaryReaderError> {
-    let mut imports = Vec::new();
-    let mut allocator_trigger = false;
-    let mut ei_check = false;
-    let mut memory_grow_flag = false;
-    let mut has_panic = "none";
+    let mut wasm_info = WasmInfo::default();
+    let mut write_functions: HashSet<usize> = HashSet::new();
 
     let parser = Parser::new(0);
     for payload in parser.parse_all(&wasm_data) {
         match payload? {
             Payload::ImportSection(import_section) => {
-                imports = extract_imports(import_section, extract_imports_enabled);
-                ei_check = is_ei_valid(imports.clone(), check_ei);
+                write_functions =
+                    process_imports(import_section, extract_imports_enabled, &mut wasm_info);
+                wasm_info.ei_check = is_ei_valid(&wasm_info.imports, check_ei);
             },
             Payload::DataSection(data_section) => {
-                allocator_trigger = is_fail_allocator_triggered(data_section.clone());
+                wasm_info.report.has_allocator = is_fail_allocator_triggered(data_section.clone());
                 if is_panic_with_message_triggered(data_section.clone()) {
-                    has_panic = WITH_MESSAGE;
+                    WITH_MESSAGE.clone_into(&mut wasm_info.report.has_panic);
                 } else if is_panic_without_message_triggered(data_section) {
-                    has_panic = WITHOUT_MESSAGE;
+                    WITHOUT_MESSAGE.clone_into(&mut wasm_info.report.has_panic);
                 }
             },
             Payload::CodeSectionEntry(code_section) => {
-                memory_grow_flag = is_mem_grow(code_section);
+                wasm_info.memory_grow_flag = is_mem_grow(&code_section);
+                create_call_graph(code_section, &mut wasm_info.call_graph);
+            },
+            Payload::ExportSection(export_section) => {
+                parse_export_section(export_section, &mut view_endpoints);
             },
             _ => (),
         }
     }
 
+    detect_write_operations_in_views(&view_endpoints, &wasm_info.call_graph, &mut write_functions);
+
     let report = ReportCreator {
         path,
-        has_allocator: allocator_trigger,
-        has_panic: has_panic.to_string(),
+        has_allocator: wasm_info.report.has_allocator,
+        has_panic: wasm_info.report.has_panic,
     };
 
     Ok(WasmInfo {
-        imports,
-        ei_check,
-        memory_grow_flag,
-        has_format: true,
+        imports: wasm_info.imports,
+        ei_check: wasm_info.ei_check,
+        memory_grow_flag: wasm_info.memory_grow_flag,
+        call_graph: wasm_info.call_graph,
         report,
     })
 }
 
+fn detect_write_operations_in_views(
```

Comment:
It would be nice to also provide a unit test. It shouldn't be too difficult, you can embed the adder as a WAT.